src/test/java/org/codehaus/stax/test/stream/TestEntityRead.java

```diff
@@ -407,6 +457,35 @@ public void testInvalidEntityPartial()
         assertTokenType(START_ELEMENT, type2);
         fail("Expected an exception for partial entity reference: current token after text: "+tokenTypeDesc(lastType));
     }
+    
+    /**
+     * Test that ensures that an invalid surrogate pair entities is caught.
+     * It could be pair of high surrogate and simple entity, high surrogate
+     * with no pair, low surrogate as first or unclosed entity
+     */
+    public void testInvalidSurrogatePairEntities()
+            throws XMLStreamException
+    {
+        final String[] invalidSurrogatePairs = {
+            // Invalid pair
+            "<root>surrogate pair: &#55356;&#5722;.</root>",
+            // No pair
+            "<root>surrogate pair: &#55356;.</root>",
+            // Low surrogate as first
+            "<root>surrogate pair: &#57221;&#55356;.</root>",
+            // Unclosed second entity
+            "<root>surrogate pair: &#55356;&#553</root>"
+        };
+        
+        for (String invalidSurrogatePair: invalidSurrogatePairs) {
+            XMLStreamReader sr = getReader(invalidSurrogatePair,
+                    true, false, true, true);
+            try {
+                streamThrough(sr);
+                fail("Expected an exception for invalid surrogate pair");
+            } catch (Exception e) { }
```

Comment:
Would be better to catch specific `Exception` subtype and check exception message that it fails for expected reason (just match a phrase, no need to match exact message).